### PR TITLE
Add `fetch-commit` command

### LINKED ISSUE #162: interface to re-fetch logs for some specific PR
in attempt to troubleshoot https://github.com/datalad/datalad/issues/7296 I thought to compare logs between to CI runs on that PR, for which we should have had logs on smaug a264136 and 2340b4c, for which we have

```
$> ls -ld */pr/7271/{a264136,2340b4c} 
drwxr-sr-x 1 datalad datalad 930 Feb 22 03:28 21/pr/7271/2340b4c/
drwxr-sr-x 1 datalad datalad 166 Feb 22 03:20 21/pr/7271/a264136/
drwxr-sr-x 1 datalad datalad  92 Feb 23 03:20 22/pr/7271/2340b4c/
```

<details>
<summary>placing the oddity of 1 fastest github run getting into 22nd aside</summary> 

```shell
(git)smaug:/mnt/datasets/datalad/ci/logs/2023/02[master]git
$> ls -l 22/pr/7271/2340b4c/ 21/pr/7271/2340b4c/
21/pr/7271/2340b4c/:
total 0
drwxr-sr-x 1 datalad datalad 280 Feb 22 03:28 appveyor-9134-failed/
drwxr-sr-x 1 datalad datalad 120 Feb 22 03:28 appveyor-9134-success/
drwxr-sr-x 1 datalad datalad  24 Feb 22 03:20 github-Add\ changelog.d\ snippet-565-incomplete/
drwxr-sr-x 1 datalad datalad  86 Feb 22 03:20 github-Benchmarks-6132-success/
drwxr-sr-x 1 datalad datalad  56 Feb 22 03:20 github-CrippledFS-3876-success/
drwxr-sr-x 1 datalad datalad  62 Feb 22 03:20 github-Docs-6191-success/
drwxr-sr-x 1 datalad datalad 648 Feb 22 03:20 github-Extensions-6140-failed/
drwxr-sr-x 1 datalad datalad  56 Feb 22 03:20 github-Linters-3027-success/
drwxr-sr-x 1 datalad datalad  56 Feb 22 03:20 github-Shellcheck\ on\ scripts-4379-success/
drwxr-sr-x 1 datalad datalad  56 Feb 22 03:20 github-Test\ for\ semver\ label-3182-success/
drwxr-sr-x 1 datalad datalad  56 Feb 22 03:20 github-Test\ old\ nose\ code-795-success/
drwxr-sr-x 1 datalad datalad 148 Feb 22 03:20 github-Test\ on\ macOS-4368-failed/
drwxr-sr-x 1 datalad datalad  68 Feb 22 03:20 github-Type-check-821-success/
drwxr-sr-x 1 datalad datalad  20 Feb 22 03:28 travis-15237-errored/
drwxr-sr-x 1 datalad datalad  70 Feb 22 03:28 travis-15237-failed/
drwxr-sr-x 1 datalad datalad  12 Feb 22 03:28 travis-15237-success/

22/pr/7271/2340b4c/:
total 0
drwxr-sr-x 1 datalad datalad 40 Feb 23 03:20 github-Auto-release\ on\ PR\ merge-325-incomplete/

```
</details>

what is most upsetting is that we have only few of non-interesting logs for the other one:

```shell
$> ls 21/pr/7271/a264136/
github-Add\ changelog.d\ snippet-564-success/  github-Test\ for\ semver\ label-3181-success/
```

AFAIK we do not have logs for those tinuous runs anywhere so can't tell why that happened (may be the idea of #154 would still have value). So I thought that may be there is a way to have a command to (re)fetch logs for some specific PR/commit?  @jwodder -- would that be possible?

### PR DESCRIPTION
**Pull request by jwodder**
Closes #162.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -46,7 +46,6 @@ install_requires =
     click-loglevel ~= 0.2
     in_place ~= 0.4
     pydantic ~= 1.7
-    PyGithub ~= 1.55
```
Comment (yarikoptic) [inline setup.cfg:49]: I wonder why you decided to rewrite those interfaces without using pygithub?

Comment (jwodder) [reply setup.cfg:49]: Pygithub doesn't support fetching workflow run for a given commit, nor does it support several other run-related things, nor does it properly parse timestamps, and retrying requests via Pygithub when rate limits are hit was harder than it needed to be.  I think tinuous has reached the point where Pygithub is more of a hindrance than a help.

**Comment by yarikoptic** [comment]
not sure why there is no codecov report, but from https://app.codecov.io/gh/con/tinuous/pull/171 `HEAD 65.40%
Patch 42.79% Change 2.62%`.  Since coverage of the patch is relatively low I wonder if you had a chance to try this PR on some sample repos to ensure that there is no regressions?

**Comment by jwodder** [comment]
@yarikoptic I successfully ran `fetch-commit` on datalad's `maint` branch, but I did not test regular `fetch`.

**Comment by yarikoptic** [comment]
> @yarikoptic I successfully ran `fetch-commit` on datalad's `maint` branch, but I did not test regular `fetch`.

Please check on some of the existing setups (e.g. for datalad) while going a commit or two back and then doing diff on either we got the same result.

**Pull request merged**